node/pegnet/admin.go

```diff
@@ -0,0 +1,120 @@
+package pegnet
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+)
+
+// This file can be used for node administration related functions
+
+var (
+	// PegnetdSyncVersion is an indicator of the version of pegnetd
+	// at each height synced. This version number can differ from the tagged
+	// version, and is likely only to be updated at hard forks. It is used to
+	// detect if a pegnetd was updated late, and therefore has an invalid state.
+	//
+	// Each fork should increment this number by at least 1
+	PegnetdSyncVersion = 1
+)
+
+type ForkEvent struct {
+	ActivationHeight uint32
+	MinimumVersion   int
+}
+
+var (
+	Hardforks = []ForkEvent{
+		// This is the most basic check. All versions are valid for 0
+		{0, -1}, // {0, -1}, means at height 0 any version >= -1 is sufficient
+
+		// Pegnet activation. When Pegnet starts on mainnet
+		{206422, 1},
```

Comment:
how does this work for existing databases? pn_sync_version only gets populated for new sync entries but the hardfork check would expect entries for 206422 with version 1, but get -1. in order for this to work, all databases would have to either by synced from scratch or you need to autopopulate the table

the activation of eth_gateway should be hard fork 1, since it's the first one detectable by this code. everything before that is -1 (we can skip 0)

Reply:
You are right, I'll drop that. When we decide the eth_gateway fork height, I'll add it